src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -0,0 +1,212 @@
+package seedu.duke.parser;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import seedu.duke.command.ByeCommand;
+import seedu.duke.command.CheckCommand;
+import seedu.duke.command.Command;
+import seedu.duke.command.DeadlineCommand;
+import seedu.duke.command.DeleteCommand;
+import seedu.duke.command.EventCommand;
+import seedu.duke.command.FindCommand;
+import seedu.duke.command.ListCommand;
+import seedu.duke.command.MarkCommand;
+import seedu.duke.command.TodoCommand;
+import seedu.duke.command.UnmarkCommand;
+import seedu.duke.common.Messages;
+import seedu.duke.exception.InvalidCommandFormatException;
+import seedu.duke.exception.UnknownCommandException;
+
+
+/**
+ * The Parser class implements a method that would parse the user input into Command
+ */
+public class Parser {
+    public static final Pattern BASIC_COMMAND_FORMAT = Pattern.compile("(?<commandWord>\\S+)(?<arguments>.*)");
```

Comment:
Maybe can add comments on what these are suppose to represent? Just a suggestion! cause i dont understand the basic commandformat and one arg command format